src/core/parse.js

```diff
@@ -301,16 +307,22 @@ foam.CLASS({
     {
       name: 's',
       final: true
-    }
+    },
+    {
+      name: 'case',
+      value: true,
+      final: true
+    },
   ],
 
   methods: [
     function step() {
       var ps1 = this.ps;
       var ps = this.ps;
-      var str = this.s;
+      var str = this.case ? this.s : this.s.toLowerCase();
       for ( var i = 0 ; i < str.length ; i++, ps = ps.tail ) {
-        if ( str.charAt(i) !== ps.head ) {
+        if ( ! ps.head ||
+            str.charAt(i) !== (this.case ? ps.head : ps.head.toLowerCase()) ) {
```

Comment:
I'm concerned about the performance this will have the parsers, adding an extra branch point every character even if we're not in a literal_ic situation.

Let's make LiteralIC it's own parser like it was in foam1